repository/gogit.go

```diff
@@ -170,6 +171,28 @@ func detectGitPath(path string) (string, error) {
 		fi, err := os.Stat(filepath.Join(path, ".git"))
 		if err == nil {
 			if !fi.IsDir() {
+				// See if our .git item is a dotfile that holds a submodule reference
+				dotfile, err := os.Open(fi.Name())
+				if err != nil {
+					// Can't open" error
+					return "", fmt.Errorf(".git exists but is not a directory or a readable file: %w", err)
+				}
+				// We aren't going to defer the dotfile.Close, because we might keep looping, so we have to be sure to
+				// clean up before returning an error
+				reader := bufio.NewReader(dotfile)
+				line, _, err := reader.ReadLine()
+				if err != nil {
+					_ = dotfile.Close()
+					return "", fmt.Errorf(".git exists but is not a direcctory and cannot be read: %w", err)
+				}
+				dotContent := string(line)
+				if strings.HasPrefix(dotContent, "gitdir:") {
+					_ = dotfile.Close()
```

Comment:
This `Close()` is misplaced, you want to close as soon as possible (just after the `ReadLine()` here, after handling the error). In your code still have one code path that can lead to not closing the file.